internal/org/resolvers/external.go

```diff
@@ -0,0 +1,154 @@
+package resolvers
+
+import (
+	"context"
+	"time"
+
+	"github.com/flightctl/flightctl/internal/auth/common"
+	"github.com/flightctl/flightctl/internal/org/providers"
+	"github.com/flightctl/flightctl/internal/store/model"
+	"github.com/google/uuid"
+	"github.com/jellydator/ttlcache/v3"
+	"github.com/sirupsen/logrus"
+)
+
+// ExternalResolver caches organization ID validation.
+type ExternalResolver struct {
+	store               OrgStore
+	cache               *ttlcache.Cache[uuid.UUID, *model.Organization]
+	ttl                 time.Duration
+	externalOrgProvider providers.ExternalOrganizationProvider
+	log                 logrus.FieldLogger
+}
+
+// NewExternalResolver constructs a new resolver. A TTL of zero disables expiration.
+func NewExternalResolver(ctx context.Context, s OrgStore, ttl time.Duration, externalOrgProvider providers.ExternalOrganizationProvider, log logrus.FieldLogger) *ExternalResolver {
+	opts := []ttlcache.Option[uuid.UUID, *model.Organization]{}
+	if ttl > 0 {
+		opts = append(opts, ttlcache.WithTTL[uuid.UUID, *model.Organization](ttl))
+	}
+	c := ttlcache.New(opts...)
+
+	go func() {
+		c.Start()
+		<-ctx.Done()
+		c.Stop()
+	}()
+
+	return &ExternalResolver{
+		store:               s,
+		cache:               c,
+		ttl:                 ttl,
+		externalOrgProvider: externalOrgProvider,
+		log:                 log,
+	}
+}
+
+func (r *ExternalResolver) EnsureExists(ctx context.Context, id uuid.UUID) error {
+	_, err := r.getOrg(ctx, id)
+	return err
+}
+
+// getOrg fetches the organization for the given ID. It caches lookups according
+// to the configured TTL. Failed lookups are not cached, ensuring that newly
+// created organizations are immediately accessible.
+func (r *ExternalResolver) getOrg(ctx context.Context, id uuid.UUID) (*model.Organization, error) {
+	if item := r.cache.Get(id); item != nil {
+		return item.Value(), nil
+	}
+	// Cache miss – query the store.
+	org, err := r.store.GetByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use configured TTL or disable expiration if TTL is not positive
+	cacheTTL := ttlcache.NoTTL
+	if r.ttl > 0 {
+		cacheTTL = r.ttl
+	}
+	r.cache.Set(id, org, cacheTTL)
+	return org, nil
+}
+
+// Close stops the cache and releases resources. Should be called when the resolver
+// is no longer needed to prevent goroutine leaks.
+func (r *ExternalResolver) Close() {
+	r.cache.Stop()
+}
+
+func (r *ExternalResolver) IsMemberOf(ctx context.Context, identity common.Identity, id uuid.UUID) (bool, error) {
+	org, err := r.getOrg(ctx, id)
+	if err != nil {
+		return false, err
+	}
+
+	return r.externalOrgProvider.IsMemberOf(ctx, identity, org.ExternalID)
+}
+
+func (r *ExternalResolver) GetUserOrganizations(ctx context.Context, identity common.Identity) ([]*model.Organization, error) {
```

Comment:
I'm probably missing something, but I'm a little confused by the overall flow. It appears that external -> internal org mapping/creation is done in this call, but this is only referenced from `service.ListOrganizations`. If this isn't called, regular API requests will fail as the org doesn't exist in the store. How are we ensuring that List Organizations is called?